programs/bench.c

```diff
@@ -647,96 +901,80 @@ static void BMK_benchFileTable(const char* const * const fileNamesTable, unsigne
-            BMK_benchCLevel(srcPtr, fileSize,
-                            fileSizes+fileNb, 1, 
-                            cLevel, cLevelLast, compressionParams,
-                            dictBuffer, dictBufferSize, 
-                            displayLevel, fileNamesTable[fileNb]);
-            srcPtr += fileSize;
+    {
+        int errorCode = BMK_loadFiles(srcBuffer, benchedSize, fileSizes, fileNamesTable, nbFiles, displayLevel);
+        if(errorCode) {
+            res.error = errorCode;
+            return res;
         }
-
-    } else {
+    }
+    /* Bench */
+    {
         char mfName[20] = {0};
         snprintf (mfName, sizeof(mfName), " %u files", nbFiles);
-        {   const char* const displayName = (nbFiles > 1) ? mfName : fileNamesTable[0];
-            BMK_benchCLevel(srcBuffer, benchedSize, 
+        {   
+            const char* const displayName = (nbFiles > 1) ? mfName : fileNamesTable[0];
+            res = BMK_benchCLevel(srcBuffer, benchedSize, 
                             fileSizes, nbFiles, 
-                            cLevel, cLevelLast, compressionParams,
+                            cLevel, compressionParams,
                             dictBuffer, dictBufferSize, 
-                            displayLevel, displayName);
+                            displayLevel, displayName, 
+                            adv);
     }   }
 
     /* clean up */
     free(srcBuffer);
     free(dictBuffer);
     free(fileSizes);
+    return res;
 }
 
 
-static void BMK_syntheticTest(int cLevel, int cLevelLast, double compressibility,
+BMK_return_t BMK_syntheticTest(int cLevel, double compressibility,
                               const ZSTD_compressionParameters* compressionParams,
-                              int displayLevel)
+                              int displayLevel, const BMK_advancedParams_t * const adv)
 {
     char name[20] = {0};
     size_t benchedSize = 10000000;
-    void* const srcBuffer = malloc(benchedSize);
+    void* srcBuffer;
+    BMK_return_t res;
+
+    if (cLevel > ZSTD_maxCLevel()) {
+        EXM_THROW(15, BMK_return_t, "Invalid Compression Level");
+    }
 
     /* Memory allocation */
-    if (!srcBuffer) EXM_THROW(21, "not enough memory");
+    srcBuffer = malloc(benchedSize);
+    if (!srcBuffer) EXM_THROW(21, BMK_return_t, "not enough memory");
 
     /* Fill input buffer */
     RDG_genBuffer(srcBuffer, benchedSize, compressibility, 0.0, 0);
 
     /* Bench */
     snprintf (name, sizeof(name), "Synthetic %2u%%", (unsigned)(compressibility*100));
-    BMK_benchCLevel(srcBuffer, benchedSize, 
+    res = BMK_benchCLevel(srcBuffer, benchedSize, 
                     &benchedSize, 1, 
-                    cLevel, cLevelLast, compressionParams, 
+                    cLevel, compressionParams, 
                     NULL, 0, 
-                    displayLevel, name);
+                    displayLevel, name, adv);
 
     /* clean up */
-    free(srcBuffer);
-}
-
+    free((void*)srcBuffer);
```

Comment:
No need to cast to `void*`.

Reply:
VS gives a 'different const qualifiers' warning on the free() call without the cast.

Reply:
I would expect that in previous function `free((void*)srcPtrs);`, due to a bug in Visual,

but here, Visual should not complain. `srcBuffer` is clearly defined as `void*`, not `const void*`.

Reply:
Oops, yeah. I accidentally deleted the wrong cast when looking at this comment last time, it should be fixed now.